src/Stratis.Bitcoin/P2P/Peer/NetworkPeerServer.cs

```diff
@@ -168,28 +168,29 @@ public void Listen(int maxIncoming = 20)
 
             this.logger.LogTrace("Accepting incoming connections.");
 
-            while (!this.serverCancel.IsCancellationRequested)
+            try
             {
-                try
+                while (!this.serverCancel.IsCancellationRequested)
                 {
-                    this.serverCancel.Token.ThrowIfCancellationRequested();
-
-                    TcpClient tcpClient = await Task.Run(async () => await this.tcpListener.AcceptTcpClientAsync(), this.serverCancel.Token);
+                    TcpClient tcpClient = await Task.Run(async () => await this.tcpListener.AcceptTcpClientAsync(), this.serverCancel.Token).ConfigureAwait(false);
                     NetworkPeerClient client = this.networkPeerFactory.CreateNetworkPeerClient(tcpClient);
 
                     this.AddConnectedClient(client);
 
                     this.logger.LogTrace("Connection accepted from client '{0}'.", client.RemoteEndPoint);
-                    Task unused = Task.Run(() => this.ProcessNewClient(client));
-                }
-                catch (Exception e)
-                {
-                    if (e is OperationCanceledException) this.logger.LogDebug("Shutdown detected, stop accepting connections.");
-                    else this.logger.LogDebug("Exception occurred: {0}");
 
-                    break;
+                    // This should be cheaper for the accept loop thread than just calling ProcessNewClientAsync without awaiting.
+                    Task unused = Task.Run(async () => await this.ProcessNewClientAsync(client));
                 }
             }
+            catch (OperationCanceledException)
+            {
+                this.logger.LogDebug("Shutdown detected, stop accepting connections.");
+            }
+            catch (Exception e)
+            {
+                this.logger.LogDebug("Exception occurred: {0}", e.ToString());
```

Comment:
`LogError` perhaps?

Reply:
i am still not sure, it seems like it is possible that the socket will be disposed and it would cause the exception, this code is still under review